src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -93,31 +154,64 @@ public RoadAccident getAccidentByIndex(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+        return roadAccidentList.stream()
+        		.filter(r -> r.getLongitude() > minLongitude).filter(r -> r.getLongitude() < maxLongitude)
+        		.filter(r -> r.getLatitude() > minLatitude).filter(r -> r.getLatitude() < maxLatitude)
+        		.collect(Collectors.toList());
     }
 
     /**
      * find the weather conditions which caused max number of incidents
      * @return
      */
     public List<String> getTopThreeWeatherCondition(){
-        return null;
+		Map<String, Integer> map = new HashMap<String, Integer>();
+		roadAccidentList.stream().map(RoadAccident::getWeatherConditions).forEach(r -> {
+			Integer count = map.get(r);
+			if (count == null)
+				map.put(r, 1);
+			else
+				map.put(r, count + 1);
+		});
+
+		return map.entrySet().stream().sorted(new Comparator<Map.Entry<String, Integer>>() {
+			@Override
+			public int compare(Entry<String, Integer> o1, Entry<String, Integer> o2) {
+				return o2.getValue() - o1.getValue();
+			}
+		}).map(Entry::getKey).collect(Collectors.toList()).subList(0, 3);
     }
 
     /**
      * count incidents by road surface conditions
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
+    	Map<String, Long> map = new HashMap<String, Long>();
+    	roadAccidentList.stream().map(RoadAccident::getRoadSurfaceConditions).forEach(r -> {
+    		Long count = map.get(r);
+    		if (count == null)
+    			map.put(r, 1L);
+    		else
+    			map.put(r, count + 1);
+    	});
+    	return map;
     }
 
     /**
      * To match streaming operations result, return type is a java collection instead of multimap
      * @return
      */
     public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
+    	Map<String, List<String>> map = new HashMap<String, List<String>>();
```

Comment:
Also - using groupingBy will do this task and previous in the one short string.

but by default groupingBy will return map with String, List<Accidents>   so need additional mapping.(inside groupingBy)